v2/util/qc_stmt.go

```diff
@@ -117,23 +147,50 @@ func checkAsn1Reencoding(i interface{}, originalEncoding []byte, appendIfCompari
 	return result
 }
 
-func IsAnyEtsiQcStatementPresent(extVal []byte) bool {
-	oidList := make([]*asn1.ObjectIdentifier, 6)
-	oidList[0] = &IdEtsiQcsQcCompliance
-	oidList[1] = &IdEtsiQcsQcLimitValue
-	oidList[2] = &IdEtsiQcsQcRetentionPeriod
-	oidList[3] = &IdEtsiQcsQcSSCD
-	oidList[4] = &IdEtsiQcsQcEuPDS
-	oidList[5] = &IdEtsiQcsQcType
-	for _, oid := range oidList {
-		r := ParseQcStatem(extVal, *oid)
-		if r.IsPresent() {
+var etsiQCStatements = []asn1.ObjectIdentifier{
+	IdEtsiQcsQcCompliance,
+	IdEtsiQcsQcLimitValue,
+	IdEtsiQcsQcRetentionPeriod,
+	IdEtsiQcsQcSSCD,
+	IdEtsiQcsQcEuPDS,
+	IdEtsiQcsQcType,
+}
+
+func IsAnyEtsiQcStatementPresent(c *x509.Certificate) bool {
+	statements := c.QCStatements
+	statementIDs := statements.StatementIDs
+	for _, oid := range statementIDs {
```

Comment:
It's worth calling out you use different syntax in Line 173.

Should this just be:
```suggestion
	for _, oid := range c.QCStatements.StatementIDs {
```